src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,153 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static User user;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user.setAge(23);
+        user.setLogin("Login123");
+        user.setPassword("qwerty123");
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_invalidLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setLogin("");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setLogin("log");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setLogin("login");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_invalidAge_NotOk() {
+        user.setAge(null);
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+        user.setAge(0);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setAge(-1);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setAge(-10);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setAge(10);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        user.setAge(16);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validAge_Ok() {
+        User userAge18 = new User();
+        userAge18.setLogin("UserAge18");
+        userAge18.setPassword("password18");
+        userAge18.setAge(18);
```

Comment:
Better save magic numbers in constant and correct naming of userValidAge or like this